homeassistant/components/advantage_air/climate.py

```diff
@@ -0,0 +1,283 @@
+    "heat": HVAC_MODE_HEAT,
+    "cool": HVAC_MODE_COOL,
+    "vent": HVAC_MODE_FAN_ONLY,
+    "dry": HVAC_MODE_DRY,
+}
+HASS_HVAC_MODES = {v: k for k, v in ADVANTAGE_AIR_HVAC_MODES.items()}
+
+ADVANTAGE_AIR_FAN_MODES = {
+    "auto": FAN_AUTO,
+    "low": FAN_LOW,
+    "medium": FAN_MEDIUM,
+    "high": FAN_HIGH,
+}
+HASS_FAN_MODES = {v: k for k, v in ADVANTAGE_AIR_FAN_MODES.items()}
+FAN_SPEEDS = {FAN_LOW: 30, FAN_MEDIUM: 60, FAN_HIGH: 100}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform setup isn't required."""
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir climate platform."""
+
+    instance = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        entities.append(AdvantageAirAC(instance, ac_key))
+        for zone_key in instance["coordinator"].data["aircons"][ac_key]["zones"]:
+            # Only add zone climate control when zone is in temperature control
+            if (
+                instance["coordinator"].data["aircons"][ac_key]["zones"][zone_key][
+                    "type"
+                ]
+                != 0
+            ):
+                entities.append(AdvantageAirZone(instance, ac_key, zone_key))
+    async_add_entities(entities)
+    return True
+
+
+class AdvantageAirClimateEntity(CoordinatorEntity, ClimateEntity):
+    """AdvantageAir Climate class."""
+
+    def __init__(self, instance):
+        """Initialize the base Advantage Air climate entity."""
+        super().__init__(instance["coordinator"])
+        self.async_change = instance["async_change"]
+        self.device = instance["device"]
+
+    @property
+    def temperature_unit(self):
+        """Return the temperature unit."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported temperature step."""
+        return PRECISION_WHOLE
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature."""
+        return 32
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature."""
+        return 16
+
+    @property
+    def device_info(self):
+        """Return parent device information."""
+        return self.device
+
+
+class AdvantageAirAC(AdvantageAirClimateEntity):
+    """AdvantageAir AC unit."""
+
+    def __init__(self, instance, ac_key):
+        """Initialize the Advantage Air AC climate entity."""
+        super().__init__(instance)
+        self.ac_key = ac_key
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["name"]
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-climate'
```

Comment:
```suggestion
        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}'
```
unique_id only have to be unique per platform so if you don't plan to introduce multiple climate entities that are based on the same data source you can remove the -climate suffix, same comment is valid for the zone entity

Reply:
Did not know this, will do.